roles/ceph-mon/tasks/rbd_pool.yml

```diff
@@ -1,17 +1,36 @@
 ---
-- name: check rbd pool usage
-  shell: |
-    ceph --connect-timeout 5 --cluster {{ cluster }} df | awk '/rbd/ {print $3}'
-  changed_when: false
-  failed_when: false
-  always_run: true
-  register: rbd_pool_df
 
-- name: check pg num for rbd pool
+- name: test if rbd exists
   shell: |
-    ceph --connect-timeout 5 --cluster {{ cluster }} osd pool get rbd pg_num | awk '{print $2}'
+    ceph --cluster {{ cluster }} osd pool ls | grep -sq rbd
   changed_when: false
   failed_when: false
-  always_run: true
-  register: rbd_pool_pgs
+  run_once: true
+  check_mode: true
+  register: rbd_pool_exist
+
+- name: include rbd_pool_df.yml
+  include: rbd_pool_df.yml
+  when: rbd_pool_exist.rc == 0
+
+- name: include rbd_pool_pgs.yml
+  include: rbd_pool_pgs.yml
+  when:
+    - rbd_pool_exist.rc == 0
+    - global_in_ceph_conf_overrides
+    - ceph_conf_overrides.global.osd_pool_default_pg_num is defined
+
+- name: include rbd_pool_size.yml
+  include: rbd_pool_size.yml
+  when:
+    - rbd_pool_exist.rc == 0
+    - global_in_ceph_conf_overrides
+    - ceph_conf_overrides.global.osd_pool_default_size is defined
 
+- name: include rbd_pool_luminous.yml
```

Comment:
I personally won't change the default behavior of Ceph, so for and after Luminous let's not creating the RBD pool.